Silicon/CommonSocPkg/Library/HeciLib/HeciLib.c

```diff
@@ -0,0 +1,243 @@
+/** @file
+  Heci driver core, determines the HECI device and initializes it.
+
+  Copyright (c) 2017 - 2020, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Base.h>
+#include <Library/BaseLib.h>
+#include <Library/IoLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/TimerLib.h>
+#include <Library/HeciLib.h>
+#include <Library/BootloaderCommonLib.h>
+#include <IndustryStandard/Pci.h>
+#include "HeciRegs.h"
+
+/**
+  Return HECI1 Mode
+
+  @param[out] MeMode              Pointer for HECI1 Mode report
+
+  @retval EFI_SUCCESS             MeMode copied
+  @retval EFI_INVALID_PARAMETER   Pointer of MeMode is invalid
+**/
+EFI_STATUS
+EFIAPI
+HeciGetMeMode (
+  OUT UINT32                     *MeMode
+  )
+{
+  HECI_FWS_REGISTER                   MeFirmwareStatus;
+  UINTN                               HeciMmPciAddress;
+
+  if (MeMode == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  HeciMmPciAddress = MeGetHeciMmPciAddress (0, 0);
+  if (MmioRead16 (HeciMmPciAddress + PCI_DEVICE_ID_OFFSET) == 0xFFFF) {
+    return EFI_NOT_FOUND;
+  }
+
+  MeFirmwareStatus.ul = MmioRead32 (HeciMmPciAddress + R_ME_HFS);
+  switch (MeFirmwareStatus.r.MeOperationMode) {
+    case ME_OPERATION_MODE_NORMAL:
+      *MeMode = ME_MODE_NORMAL;
+      break;
+
+    case ME_OPERATION_MODE_DEBUG:
+      *MeMode = ME_MODE_DEBUG;
+      break;
+
+    case ME_OPERATION_MODE_SOFT_TEMP_DISABLE:
+      *MeMode = ME_MODE_TEMP_DISABLED;
+      break;
+
+    case ME_OPERATION_MODE_SECOVR_JMPR:
+    case ME_OPERATION_MODE_SECOVR_HECI_MSG:
+      *MeMode = ME_MODE_SECOVER;
+      break;
+
+    case ME_OPERATION_MODE_SPS:
+      *MeMode = ME_MODE_SPS;
+      break;
+
+    default:
+      *MeMode = ME_MODE_FAILED;
+  }
+  DEBUG ((DEBUG_INFO, "[HECI1] MeMode %d (Hfsts1: %08X)\n", *MeMode, MeFirmwareStatus.ul));

```

Comment:
This is getting printed lot of times in firmware update, can we make it verbose statement?

Reply:
Let me remove the debug message here. I will change all debug level in HeciCore.c to DEBUG_VERBOSE.